### PR TITLE
Add new reasons param to react-autosuggest ShouldRenderSuggestions

### PR DESCRIPTION
**Pull request by pimterry**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/moroshko/react-autosuggest/pull/773
- [ ] ~~If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header~~ Brings up to date with 10.0.3, but patch versions aren't allowed in the header
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [ ] ~~If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.~~ already in place

### CODE REVIEW

File: types/react-autosuggest/index.d.ts [hunk 1]
```diff
@@ -128,7 +128,17 @@ declare namespace Autosuggest {
...
+        value: string,
+        reason:
+            | 'input-changed'
+            | 'input-focused'
+            | 'input-blurred'
+            | 'escape-pressed'
+            | 'suggestions-revealed'
+            | 'suggestions-updated'
+            | 'render'
+    ) => boolean;
```
Comment (cdeutsch) [inline types/react-autosuggest/index.d.ts:141]: Maybe make `Reason` a type instead of doing it in line?

Otherwise looks good 👍

**Review by cdeutsch — CHANGES_REQUESTED** [review]

File: types/react-autosuggest/index.d.ts (hunk 1 above)
Comment (pimterry) [reply types/react-autosuggest/index.d.ts]: Good idea! I've extracted both this type for reasons here and for FetchRequest (overlapping, but slightly different) into standalone types.

**Review by cdeutsch — APPROVED** [review]
Looks good 👍

File: types/react-autosuggest/index.d.ts (hunk 1 above)
Comment (cdeutsch) [reply types/react-autosuggest/index.d.ts]: Nice!

Thanks for doing that

**Comment by pimterry** [comment]
Hmm, I think this is all good to merge, and the bot should've marked it as 'Owner Approved' and offered to merge now, but it hasn't... Any ideas? Maybe a comment like this will nudge it onwards :smile:

**Comment by pimterry** [comment]
@sandersn can you or another DT maintainer give me a hand here? This PR is all green and approved by an owner, but the bot doesn't seem to have noticed.

**Review by cdeutsch — APPROVED** [review]

**Comment by cdeutsch** [comment]
Ready to merge

**Comment by pimterry** [comment]
Thanks @cdeutsch!

**Pull request merged**